src/settings.cpp

```diff
@@ -1262,6 +1264,27 @@ void LoadFromConfig(bool startup)
 			}
 		}
 
+		if (generic_version < IFV_AUTOSAVE_RENAME) {
+			IniGroup *gui = generic_ini.GetGroup("gui", false);
+			if (gui != nullptr) {
+				IniItem *autosave = gui->GetItem("autosave");
+				if (autosave != nullptr) {
+					/* Convert the old settings that were in game-time to something comparable in real-time. */
+					static std::vector<std::string> _old_autosave_interval{"off", "monthly", "quarterly", "half year", "yearly"};
+					auto old_value = OneOfManySettingDesc::ParseSingleValue(autosave->value->c_str(), autosave->value->size(), _old_autosave_interval);
+
+					switch (old_value) {
+						case 0: _settings_client.gui.autosave_interval = std::chrono::minutes::zero(); break;
+						case 1: _settings_client.gui.autosave_interval = std::chrono::minutes(10); break;
+						case 2: _settings_client.gui.autosave_interval = std::chrono::minutes(30); break;
+						case 3: _settings_client.gui.autosave_interval = std::chrono::minutes(60); break;
+						case 4: _settings_client.gui.autosave_interval = std::chrono::minutes(120); break;
+						default: break;
+					}
+				}
+			}
+		}
```

Comment:
Should we check if the new setting already exists?

Not sure, but I think if you load a new (and change the value) then old then new version it will reset your `autosave_interval` to whatever `autosave` maps to?

Reply:
Yes and no. It depends :D

Configuration-files have a version, and this check ofc validates if you are loading a configuration with an older version. So what can happen is:

- You have a v4 config.
- You load a client with v5 support. It converts the `autosave` from v4 to v5.
- You now load a client with v4 support. It will resave the config as a v4.
- Now if you load a client with v5 support again, it will convert the `autosave` again, as that is what the version asks.

I am not sure this is a bad thing. How ever you slice it, if people use the same configuration with multiple versions, you will always have these weird edge-cases to resolve. This is the simplest way to approach it. What it means that if someone changes the autosave in the v4 config, it will always have an effect on v5.

If you would be to check if `autosave_interval` already exists, it won't convert it again. This can be good .. this can be bad.

So far, for all the other settings, we opted for the first. The setting in the older config wins from the newer.

Reply:
Problem with this is it will override the setting in the new version even if you didn't change the setting in the old version.

And I do often load an old version just to check behaviour of something else.

So I'm not sure I like the current way...

Reply:
Takes a bit more code, so I generalized the approach a bit. There was one other setting that did the same, so converted that. That makes this PR a bit bigger than the exact context of the PR, but helps future-us in not having this talk again to figure out what we actually want :)